Validation/L1T/plugins/L1Validator.cc

```diff
@@ -82,78 +81,141 @@ void L1Validator::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetu
   using namespace reco;
 
   Handle<GenParticleCollection> GenParticles;
-  Handle<L1EmParticleCollection> IsoEGs;
-  Handle<L1EmParticleCollection> NonIsoEGs;
-  Handle<L1JetParticleCollection> CenJets;
-  Handle<L1JetParticleCollection> ForJets;
-  Handle<L1JetParticleCollection> Taus;
-  Handle<L1MuonParticleCollection> Muons;
-  //Handle<L1EtMissParticleCollection> METs;
+  Handle<l1t::MuonBxCollection> MuonsBX;
+  Handle<l1t::EGammaBxCollection> EGammasBX;
+  Handle<l1t::TauBxCollection> TausBX;
+  Handle<l1t::JetBxCollection> JetsBX;
+  Handle<GenEventInfoProduct> genEvtInfoProduct;
+  Handle<reco::GenJetCollection> GenJets;
 
   bool GotEverything=true;
 
   if(!iEvent.getByToken(_GenSource, GenParticles)) GotEverything=false;
-  if(!iEvent.getByToken(_L1ExtraIsoEGSource, IsoEGs)) GotEverything=false;
-  if(!iEvent.getByToken(_L1ExtraNonIsoEGSource, NonIsoEGs)) GotEverything=false;
-  if(!iEvent.getByToken(_L1ExtraCenJetSource, CenJets)) GotEverything=false;
-  if(!iEvent.getByToken(_L1ExtraForJetSource, ForJets)) GotEverything=false;
-  if(!iEvent.getByToken(_L1ExtraTauJetSource, Taus)) GotEverything=false;
-  if(!iEvent.getByToken(_L1ExtraMuonSource, Muons)) GotEverything=false;
+  if(!iEvent.getByToken(_L1MuonBXSource, MuonsBX)) GotEverything=false;
+  if(!iEvent.getByToken(_L1EGammaBXSource, EGammasBX)) GotEverything=false;
+  if(!iEvent.getByToken(_L1TauBXSource, TausBX)) GotEverything=false;
+  if(!iEvent.getByToken(_L1JetBXSource, JetsBX)) GotEverything=false;
+  if(!iEvent.getByToken(_srcToken, genEvtInfoProduct)) GotEverything=false;
+  if(!iEvent.getByToken(_L1GenJetSource, GenJets)) GotEverything=false;  
 
   if(!GotEverything) return;
 
+  /*
+  std::string moduleName = "";
+  if( genEvtInfoProduct.isValid() ) {
+	  const edm::Provenance& prov = iEvent.getProvenance(genEvtInfoProduct.id());
+	  moduleName = edm::moduleName(prov);
+	  //cout<<" generator name: "<<moduleName<<endl;
+  }
+  */ 
 
   _Hists.NEvents++;
 
-  _Hists.FillNumber(L1ValidatorHists::Type::IsoEG, IsoEGs->size());
-  _Hists.FillNumber(L1ValidatorHists::Type::NonIsoEG, NonIsoEGs->size());
-  _Hists.FillNumber(L1ValidatorHists::Type::CenJet, CenJets->size());
-  _Hists.FillNumber(L1ValidatorHists::Type::ForJet, ForJets->size());
-  _Hists.FillNumber(L1ValidatorHists::Type::TauJet, Taus->size());
-  _Hists.FillNumber(L1ValidatorHists::Type::Muon, Muons->size());
+  _Hists.FillNumber(L1ValidatorHists::Type::Muon, MuonsBX->size());
+  _Hists.FillNumber(L1ValidatorHists::Type::Egamma, EGammasBX->size());
+  _Hists.FillNumber(L1ValidatorHists::Type::Tau, TausBX->size());
+  _Hists.FillNumber(L1ValidatorHists::Type::Jet, JetsBX->size());
+
+  //For gen jet
+
+  for(uint i=0; i < GenJets->size(); i++){
```

Comment:
@ly615 - please use

for ( auto &Genjet: GenJets ) { 
}

(and line 122 can be removed)